src/main/java/reactor/netty/channel/FluxReceive.java

```diff
@@ -254,7 +254,7 @@ final void drainReceiver() {
 				receiverFastpath = true;
 			}
 
-			if ((receiverDemand -= e) > 0L || e > 0L) {
+			if ((receiverDemand -= e) > 0L || (e > 0L && q.isEmpty())) {
```

Comment:
```suggestion
			if ((receiverDemand -= e) > 0L || (e > 0L && q.size() > 8)) {
```

Rather than let the queue go empty, it could proactively switch to reading below a certain level.

Reply:
By the look of it the receive window will be opened and closed on and off – we will face  a jerky transmission which will push back the overall transfer rate. I agree with @rstoyanchev it is better to have something in the wire constantly. Notwithstanding as a quick fix its ok and I would prefer to get it. 

One thought if I may: from my point of view the step change would be to control the memory consumption grand total among all active channels. In this case FluxReceive could degrade gracefully restraining the transmission if there are multiple simultaneous uploads or accept data at the highest rate. It implies the utilization of the overall quota for all concurrent channels – the implementation may be tricky.

Reply:
I created an issue about revising the FluxReceive queue low limit #1064

Reply:
Great! Thanks!